### ISSUE TITLE
Sites Management Dashboard: load data for sites

### ISSUE THREAD

**Issue by vindl** [Automattic/wp-calypso#65167]
Depends on: https://github.com/Automattic/wp-calypso/issues/65166

Let's replace the dummy data created in https://github.com/Automattic/wp-calypso/issues/65166 with the actual data. We can rely on existing data that is used to populate the Site Switcher list. Let's leave the `Last Publish` column for a separate task.

<img width="833" alt="Screenshot 2022-07-01 at 22 39 43" src="https://user-images.githubusercontent.com/1182160/176971835-86e0e85b-8e4a-48dc-8a1e-10cd2f22a60e.png">

Done is:
- [x] All of the users' sites are shown in the list e.g. Simple, Atomic, Jetpack, sites they are a member of but not the owner
- [x] The actual data is shown for the Site Icon, Site Title, Domain, Status (public, private, etc.), and Plan.

Not required:
- `Last Publish` can still use dummy data.
- Tab filtering does not need to be connected.

**Comment by danielbachhuber** [comment]
@vindl @p-jackson I think I've added all of this live data as I've worked on #65166... this issue is probably safe to close.

**Comment by vindl** [comment]
Sounds good. 👍
